tests/model_hub_tests/models_hub_common/test_performance_model.py

```diff
@@ -0,0 +1,200 @@
+# to avoid sporadic issues in inference results
+rng = np.random.default_rng(seed=56190)
+
+type_map = {
+    ov.Type.f64: np.float64,
+    ov.Type.f32: np.float32,
+    ov.Type.i8: np.int8,
+    ov.Type.i16: np.int16,
+    ov.Type.i32: np.int32,
+    ov.Type.i64: np.int64,
+    ov.Type.u8: np.uint8,
+    ov.Type.u16: np.uint16,
+    ov.Type.boolean: bool,
+}
+
+
+class Status(Enum):
+    OK = 0
+    LARGE_INFER_TIME_DIFF = 1
+    LOAD_MODEL = 2
+    GET_INPUTS_INFO = 3
+    PREPARE_INPUTS = 4
+    GET_CONVERTED_MODEL = 5
+    GET_READ_MODEL = 6
+    INFER_CONVERTED_MODEL = 7
+    INFER_READ_MODEL = 8
+    LARGE_INFER_TIME_DIFF_WITH_LARGE_VAR = 9
+
+
+class Results:
+    def __init__(self):
+        self.converted_infer_time = 0.0
+        self.converted_model_time_variance = 0.0
+        self.read_model_infer_time = 0.0
+        self.read_model_infer_time_variance = 0.0
+        self.infer_time_ratio = 0.0
+        self.error_message = ''
+        self.status = None
+
+
+def wrap_timer(func, args):
+    t0 = time.time()
+    retval = func(*args)
+    t1 = time.time()
+    return retval, t1 - t0
+
+
+class TestPerformanceModel:
+    infer_timeout = 600
+    threshold_ratio = 0.1
+    num_heat_runs = 100
+    num_measure_runs = 500
+    threshold_var = 10.0
+
+    def load_model(self, model_name, model_link):
+        raise "load_model is not implemented"
+
+    def get_inputs_info(self, model_obj):
+        raise "get_inputs_info is not implemented"
+
+    def prepare_input(self, input_shape, input_type):
+        if input_type in [ov.Type.f32, ov.Type.f64]:
+            return 2.0 * rng.random(size=input_shape, dtype=type_map[input_type])
+        elif input_type in [ov.Type.u8, ov.Type.u16, ov.Type.i8, ov.Type.i16, ov.Type.i32, ov.Type.i64]:
+            return rng.integers(0, 5, size=input_shape).astype(type_map[input_type])
+        elif input_type in [str]:
+            return np.broadcast_to("Some string", input_shape)
+        elif input_type in [bool]:
+            return rng.integers(0, 2, size=input_shape).astype(type_map[input_type])
+        else:
+            assert False, "Unsupported type {}".format(input_type)
+
+    def prepare_inputs(self, inputs_info):
+        if len(inputs_info) > 0 and inputs_info[0] == 'list':
+            inputs = []
+            inputs_info = inputs_info[1:]
+            for input_name, input_shape, input_type in inputs_info:
+                inputs.append(self.prepare_input(input_shape, input_type))
+        else:
+            inputs = {}
+            for input_name, input_shape, input_type in inputs_info:
+                inputs[input_name] = self.prepare_input(input_shape, input_type)
+        return inputs
+
+    def get_converted_model(self, model_path: str):
+        raise "get_converted_model is not implemented"
+
+    def get_read_model(self, model_path: str):
+        raise "get_read_model is not implemented"
+
+    def infer_model(self, ov_model, inputs):
```

Comment:
the base shouldn't have ov specific if we want to re-use it for tensorflow etc
we can fix it in the next PR

Reply:
After the comment @rkazants I moved here another code using ov::Model. We can discuss it later offline